Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs

```diff
@@ -12,4 +17,126 @@ void MaxTTL(EndpointConfiguration endpointConfiguration)
 
         #endregion
     }
+
+    void BucketForLargeMessages(EndpointConfiguration endpointConfiguration)
+    {
+        #region 3to4_S3BucketForLargeMessages
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        var s3Configuration = transport.S3("bucketname", "keyprefix");
+
+        #endregion
+    }
+
+    void S3Proxy(EndpointConfiguration endpointConfiguration, string bucketName, string keyPrefix)
+    {
+        #region 3to4_S3Proxy
+
+        var userName = Environment.GetEnvironmentVariable("NSERVICEBUS_AMAZONSQS_PROXY_AUTHENTICATION_USERNAME");
+        var password = Environment.GetEnvironmentVariable("NSERVICEBUS_AMAZONSQS_PROXY_AUTHENTICATION_PASSWORD");
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        var s3Configuration = transport.S3(bucketName, keyPrefix);
+        s3Configuration.ClientFactory(() => new AmazonS3Client(
+            new AmazonS3Config { 
+                ProxyCredentials = new NetworkCredential(userName, password),
+                ProxyHost = "127.0.0.1", 
+                ProxyPort = 8888 
+            }));
+
+        #endregion
+    }
+
+    void S3Region(EndpointConfiguration endpointConfiguration, string bucketName, string keyPrefix)
+    {
+        #region 3to4_S3Region
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        var s3Configuration = transport.S3(bucketName, keyPrefix);
+        s3Configuration.ClientFactory(() => new AmazonS3Client(
+            new AmazonS3Config { 
+                RegionEndpoint = RegionEndpoint.APSoutheast2
+            }));
+
+        #endregion
+    }
+
+    void S3CredentialSource(EndpointConfiguration endpointConfiguration, string bucketName, string keyPrefix)
+    {
+        #region 3to4_S3CredentialSource
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        
+        // picks up credentials as specified by the Amazon SDK
+        var s3Configuration = transport.S3(bucketName, keyPrefix);
+
+        #endregion
+    }
+
+    void S3CredentialSourceManual(EndpointConfiguration endpointConfiguration, string bucketName, string keyPrefix)
+    {
+        #region 3to4_S3CredentialSourceManual
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        var s3Configuration = transport.S3(bucketName, keyPrefix);
+        // SqsCredentialSource.InstanceProfile
```

Comment:
What's the purpose of this comment?

Reply:
The purpose of this comment is to visualize the previous enum that they had to select. So a mental mapping so to speak